src/Authentication/AnonymousUserAuthenticator.php

```diff
@@ -0,0 +1,115 @@
+<?php declare(strict_types=1);
+
+namespace Firesphere\GraphQLJWT\Authentication;
+
+use BadMethodCallException;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\ORM\ValidationException;
+use SilverStripe\ORM\ValidationResult;
+use SilverStripe\Security\Authenticator;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\MemberAuthenticator\MemberAuthenticator;
+
+class AnonymousUserAuthenticator extends MemberAuthenticator
+{
+    use Configurable;
+    use Injectable;
+
+    /**
+     * Anonymous username
+     *
+     * @var string
+     */
+    private static $anonymous_username = 'anonymous';
+
+    /**
+     * Default field values to assign to anonymous user
+     *
+     * @var array
+     */
+    private static $anonymous_fields = [
+        'FirstName' => 'Anonymous',
+    ];
+
+    public function supportedServices(): int
+    {
+        return Authenticator::LOGIN | Authenticator::LOGOUT;
+    }
+
+    public function authenticate(array $data, HTTPRequest $request, ValidationResult &$result = null): ?Member
+    {
+        // Only applies to request for anonymous user specifically
+        $email = $data['Email'] ?? null;
+        if ($email !== static::config()->get('anonymous_username')) {
+            return null;
+        }
+
+        return parent::authenticate($data, $request, $result);
+    }
+
+    /**
+     * Attempt to find and authenticate member if possible from the given data
+     *
+     * @skipUpgrade
+     * @param array            $data Form submitted data
+     * @param ValidationResult $result
+     * @param Member           $member This third parameter is used in the CMSAuthenticator(s)
+     * @return Member Found member, regardless of successful login
+     * @throws ValidationException
+     */
+    protected function authenticateMember($data, ValidationResult &$result = null, Member $member = null): Member
+    {
+        // Get user, or create if not exists
+        $member = $this->getOrCreateAnonymousMember();
```

Comment:
Would/could this possibly be a service instead of a method on `$this`? I'm aware my original wasn't that clean either, but might as well do it entirely right from this refactor

Reply:
Yes a factory wouldn't hurt.

Reply:
Pushed up.